### PR TITLE
Kadalu Content APIs SDK Updates

### PR DESCRIPTION
**Pull request by vatsa287**
Signed-off-by: Shree Vatsa N <vatsa@kadalu.tech>

### CODE REVIEW

File: python/kadalu_content_apis/objects.py [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-        self.bucket_name = bucket_name
+        self.folder_name = folder_name
         self.path = path
 
 
     @classmethod
-    def create(cls, conn, bucket_name, path, data, object_type, immutable, version, lock, template):
-        """ Create object of both default("/") and with bucket-name """
+    def create(cls, conn, folder_name, path, data, object_type, immutable, version, lock, template):
+        """ Create object of both default("/") and with folder-name """
```
Comment (aravindavk) [inline python/kadalu_content_apis/objects.py:16]: `folder_name = folder_name.lstrip("/")` and then compare `if folder_name == ""`

File: python/kadalu_content_apis/objects.py [hunk 2]
```diff
@@ -34,15 +34,15 @@ def create(cls, conn, bucket_name, path, data, object_type, immutable, version,
...
 
     @classmethod
-    def upload(cls, conn, bucket_name, file_path, object_type, path, immutable, version, lock, template):
+    def upload(cls, conn, folder_name, file_path, object_type, path, immutable, version, lock, template):
         """ Upload object data at file_path """
 
         file_content = ""
 
-        if bucket_name == "/":
+        if folder_name == "/":
```
Comment (aravindavk) [inline python/kadalu_content_apis/objects.py:42]: same here

File: python/kadalu_content_apis/objects.py [hunk 3]
```diff
@@ -68,40 +68,40 @@ def upload(cls, conn, bucket_name, file_path, object_type, path, immutable, vers
 
 
     @classmethod
-    def list(cls, conn, bucket_name):
-        """ List object(s) of both default("/") and with bucket-name """
+    def list(cls, conn, folder_name):
+        """ List object(s) of both default("/") and with folder-name """
 
-        if bucket_name == "/":
+        if folder_name == "/":
```
Comment (aravindavk) [inline python/kadalu_content_apis/objects.py:74]: same here, else two slashes comes in Line 77

Comment (vatsa287) [reply python/kadalu_content_apis/objects.py]: Updated

File: python/kadalu_content_apis/objects.py (hunk 1 above)
Comment (vatsa287) [reply python/kadalu_content_apis/objects.py:16]: done

File: python/kadalu_content_apis/folders.py [hunk 4]
```diff
@@ -65,17 +65,17 @@ def update(self, name=None, region=None, immutable=None, version=None, lock=None
...
 
-        return response_object_or_error("Bucket", resp, 200)
+        return response_object_or_error("folder", resp, 200)
 
 
-    def delete(self):
-        """ Delete buckets """
-        resp = self.conn.http_delete(f"{self.conn.url}/api/buckets/{self.name}")
-        return response_object_or_error("Bucket", resp, 204)
+    def delete(self, recursive=""):
```
[the remark this answers is not in the document]
Comment (vatsa287) [reply python/kadalu_content_apis/folders.py]: done

**Review by aravindavk — APPROVED** [review]

**Pull request merged**